File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int[] mins = new int[nums.length];
        for(int i=0;i<nums.length;i++){
            for(int j=0;j<i;j++){
                if(nums[i]>nums[j]){
                    if(mins[i]>mins[j]){
                        mins[i] = mins[i];
                    }else{
                        mins[i] = 1 + mins[j];
                    }
                }
            }
        }
        
        int max = 0;
        for(int x=0;x<nums.length;x++){
            if(max<mins[x]){
                max = mins[x];
            }
            
        }
        return max+1;
        
    }
    
}

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.py
class Solution:
    
    
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [1]*len(nums)
        for end in range(len(nums)):
            for start in range(0, end):
                if nums[end]>nums[start]:
                    dp[end]=max(dp[end],1+dp[start])
                    
        return max(dp)